### PR TITLE
Feat: Add chart of accounts type when importing accounts to category in Fyle

### PR DESCRIPTION
**Pull request by Ashutosh619-sudo**
### Description
Please add PR description here, add screenshots if needed

## Clickup
Please add link here
https://app.clickup.com/1864988/v/l/6-901604887942-1

## Release Notes

- **New Features**
  - Introduced a new field, `charts_of_accounts`, in import settings to enhance data handling.
  - Enhanced filtering capabilities for attribute imports, improving data accuracy during imports.
  - Added robust handling for account categories during data synchronization.
  - Updated import settings tests to validate the new `charts_of_accounts` functionality.

- **Bug Fixes**
  - Updated logic in import processes to ensure proper handling of new filters and parameters.

- **Documentation**
  - Added help text for the new `charts_of_accounts` field to clarify its purpose.

- **Refactor**
  - Streamlined argument handling in various import-related functions for improved clarity and efficiency.
  - Improved the `update` method in the serializer to handle changes to the `charts_of_accounts` field effectively.

### CODE REVIEW

File: apps/mappings/imports/modules/base.py [hunk 1]
```diff
@@ -57,15 +60,21 @@ def construct_attributes_filter(self, attribute_type: str, paginated_destination
...
 
         if self.sync_after and self.platform_class_name != 'expense_custom_fields':
-            filters['updated_at__gte'] = self.sync_after
+            filters["updated_at__gte"] = self.sync_after
 
         if paginated_destination_attribute_values:
-            filters['value__in'] = paginated_destination_attribute_values
+            filters["value__in"] = paginated_destination_attribute_values
+
+        account_filters = filters.copy()
```
Comment (ruuushhh) [inline apps/mappings/imports/modules/base.py:73]: why are we doing this copy thing?

File: apps/mappings/imports/modules/categories.py [hunk 2]
```diff
@@ -24,6 +27,7 @@ def trigger_import(self):
         """
         Trigger import for Category module
         """
+        print('herer')
```
Comment (ruuushhh) [inline apps/mappings/imports/modules/categories.py:30]: remove this

File: apps/mappings/imports/queues.py [hunk 3]
```diff
@@ -14,12 +14,15 @@ def chain_import_fields_to_fyle(workspace_id):
     import_settings = ImportSetting.objects.get(workspace_id=workspace_id)
     chain = Chain()
 
+    print()
```
Comment (ruuushhh) [inline apps/mappings/imports/queues.py:17]: remove this

File: apps/mappings/imports/tasks.py [hunk 4]
```diff
@@ -23,10 +24,22 @@ def trigger_import_via_schedule(workspace_id: int, destination_field: str, sourc
     import_log = ImportLog.objects.filter(workspace_id=workspace_id, attribute_type=source_field).first()
     sync_after = import_log.last_successful_run_at if import_log else None
 
+    args = {
+        'workspace_id': workspace_id,
+        'destination_field': destination_field,
+        'sync_after': sync_after,
+    }
+    
     if is_custom:
```
Comment (ruuushhh) [inline apps/mappings/imports/tasks.py:33]: two is_custom coditions are here

File: apps/mappings/imports/tasks.py [hunk 5]
```diff
@@ -23,10 +24,22 @@ def trigger_import_via_schedule(workspace_id: int, destination_field: str, sourc
...
-        item = ExpenseCustomField(workspace_id, source_field, destination_field, sync_after)
+        args['source_field'] = source_field
+    
+    if source_field == 'CATEGORY':
+        args['charts_of_accounts'] = charts_of_accounts
+
+    if is_custom:
```
Comment (ruuushhh) [inline apps/mappings/imports/tasks.py:39]: fix this

File: apps/workspaces/models.py [hunk 6]
```diff
@@ -229,6 +232,7 @@ class ImportSetting(BaseModel):
     import_categories = BooleanFalseField(help_text='toggle for import of chart of accounts from Business Central')
     import_vendors_as_merchants = BooleanFalseField(help_text='toggle for import of vendors as merchant from Business Central')
     workspace = models.OneToOneField(Workspace, on_delete=models.PROTECT, help_text='Reference to Workspace model', related_name="import_settings")
+    charts_of_accounts = ArrayField(base_field=models.CharField(max_length=100), default=get_default_chart_of_accounts, help_text='list of chart of account types to be imported into Fyle')
```
Comment (ruuushhh) [inline apps/workspaces/models.py:235]: rename to import_charts_of_accounts, just to maintain consistency

File: apps/mappings/imports/modules/base.py (hunk 1 above)
Comment (Ashutosh619-sudo) [reply apps/mappings/imports/modules/base.py:73]: we are doing this QBO as well, thought must be for some reason, like mutability

**Pull request merged**